File: frontend/src/app/data/models/action-log.ts
import { ActionType } from "../enums/actionType";
import { PropertyLog } from "./property-log";

export interface ActionLog {
    id: string,
    boardId: string,
    entityId: string,
    entityName: string,
    entityType: string,
    action: ActionType,
    creationTime: Date,
    affectedProperties: PropertyLog[]
}

File: backend/Domain/Entities/ActionLog.cs
using Domain.Common;
using Domain.Enums;

namespace Domain.Entities
{
    public class ActionLog : BaseEntity
    {
        public Guid EntityId { get; set; }
        public string EntityName { get; set; }
        public string EntityType { get; set; }
        public ActionType Action { get; set; }
        public DateTime Timestamp { get; set; }
        public IList<PropertyLog> AffectedProperties { get; set; } = new List<PropertyLog>();
    }
}
